From: Keith Busch <keith.busch@intel.com>
Date: Wed, 31 Aug 2016 14:16:01 -0600
Subject: nvme: add device id's with intel stripe quirk
Patch-mainline: Queued in NVMe legacy repository
Git-repo: git://git.infradead.org/users/kbusch/nvme-legacy.git
Git-commit: 410a5ee6f1c59dd4c1189789dd06b8ba73923b1a
References: bsc#1097562

upstream commit: 99466e708ddce8904c8635c213f2deb523ef4fb9

Back port of new intel devices requiring stripe assistance.

Signed-off-by: Keith Busch <keith.busch@intel.com>
Acked-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 drivers/block/nvme-core.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/drivers/block/nvme-core.c
+++ b/drivers/block/nvme-core.c
@@ -2603,7 +2603,10 @@ static int nvme_dev_add(struct nvme_dev
 		dev->max_hw_sectors = UINT_MAX;
 
 	if ((pdev->vendor == PCI_VENDOR_ID_INTEL) &&
-			(pdev->device == 0x0953) && ctrl->vs[3])
+			((pdev->device == 0x0953) ||
+			 (pdev->device == 0x0a53) ||
+			 (pdev->device == 0x0a54)) &&
+			ctrl->vs[3])
 		dev->stripe_size = 1 << (ctrl->vs[3] + shift);
 
 	dma_free_coherent(&dev->pci_dev->dev, 4096, mem, dma_addr);
